### PR TITLE
GH-768 AmCharts Simple & Multi Selection

### PR DESCRIPTION
**Pull request by mlzummo**
GH-768 AmCharts Simple Selection

Adds AmCharts simple selection to Bar, Funnel, Pie via selectionColor PP and selectionmode()

Fixes GH-768

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
need to figure out how to enable/disable pull out so we can do either or maybe?

File: src/amchart/Pie.js [hunk 1]
```diff
@@ -38,22 +38,28 @@
...
-    Pie.prototype.publish("useClonedPalette", false, "boolean", "Enable or disable using a cloned palette", null, { tags: ["Intermediate", "Shared"] });
+    Pie.prototype.calcRadius = function (_) {
+        return Math.min(this._size.width, this._size.height) / 2 - 2;
+    };
 
     Pie.prototype.updateChartOptions = function() {
         this._chart.type = "pie";
-        this._chart.radius = this.radius();
+        
+        this._chart.radius = this.calcRadius();
```
Comment (mlzummo) [inline src/amchart/Pie.js]: remove radius PP ?

**Comment by mlzummo** [comment]
asking amcharts if there is a way to do amchart line labels and still have piechart take up full size

**Comment by mlzummo** [comment]
how are we going to do line line/polar?

**Comment by mlzummo** [comment]
remove labelposition from pie? or atleast change it so that we can do inside and outside (line labels) ... when we figure out how to get all that working the way we like

File: demos/test-no-amd.html [hunk 2]
```diff
@@ -28,7 +28,7 @@
     <script src="../bower_components/dagre/index.js"></script>
     <script src="../bower_components/colorbrewer/colorbrewer.js"></script>
     <script src="../bower_components/topojson/topojson.js"></script>
-    <script src="../bower_components/d3-cloud/d3.layout.cloud.js"></script>
+    <script src="../bower_components/d3-cloud/build/d3.layout.cloud.js"></script>
```
Comment (GordonSmith) [inline demos/test-no-amd.html]: I have issued a PR which will fix this without you having to change here.

File: src/amchart/CommonFunnel.js [hunk 3]
```diff
@@ -126,7 +133,22 @@
...
         this._chart = AmCharts.makeChart(domNode, initObj);
-        this._chart.addListener("clickSlice", function(e) {
+        this._chart.addListener("clickSlice", function(e) {   
+            var field = e.chart.colorField;
+            var data = e.dataItem.dataContext;
+
+            if (context._selected !== null && context.selectionMode() === "simple") {
+                delete context._selected[field];
+            }
+            if (context._selected !== null && context._selected === data) {
```
Comment (GordonSmith) [inline src/amchart/CommonFunnel.js]: whould be ```else if```

File: src/amchart/CommonSerial.js [hunk 4]
```diff
@@ -499,13 +499,40 @@
         var context = this;
         var initObj = {
             type: "serial",
-            chartScrollbar: {}
+            chartScrollbar: {},
```
Comment (GordonSmith) [inline src/amchart/CommonSerial.js]: remove trailing "," (does lint not spot these?)

**Comment by GordonSmith** [comment]
Looks fine - your d3 cloud workaround is not needed.

**Comment by mlzummo** [comment]
not ready to pull ... multi selection still isnt working 100% (cant unslect)

**Comment by GordonSmith** [comment]
@mlzummo Closing until ready for pull

**Pull request closed without merging**